src/H5Ocache.c

```diff
@@ -268,53 +248,50 @@ H5O__cache_verify_chksum(const void *_image, size_t len, void *_udata)
  * Function:    H5O__cache_deserialize
  *
- * Purpose:	Attempt to deserialize the object header contained in the
- *		supplied buffer, load the data into an instance of H5O_t, and
- *		return a pointer to the new instance.
+ * Purpose:     Attempt to deserialize the object header contained in the
+ *              supplied buffer, load the data into an instance of H5O_t, and
+ *              return a pointer to the new instance.
  *
- *		Note that the object header is read with with a speculative read.
- *		If the initial read is too small, make note of this fact and return
- *     		without error.  H5C__load_entry() will note the size discrepancy
- *		and retry the deserialize operation with the correct size read.
+ *              Note that the object header is read with with a speculative
+ *              read. If the initial read is too small, make note of this fact
+ *              and return without error.  H5C__load_entry() will note the
+ *              size discrepancy and retry the deserialize operation with
+ *              the correct size read.
  *
  * Return:      Success:        Pointer to in core representation
  *              Failure:        NULL
- *
- * Programmer:  John Mainzer
- *              7/28/14
- *
  *-------------------------------------------------------------------------
  */
 static void *
 H5O__cache_deserialize(const void *image, size_t len, void *_udata, hbool_t *dirty)
 {
     H5O_t          *oh        = NULL;                     /* Object header read in */
     H5O_cache_ud_t *udata     = (H5O_cache_ud_t *)_udata; /* User data for callback */
-    void           *ret_value = NULL;                     /* Return value */
+    void           *ret_value = NULL;
 
     FUNC_ENTER_PACKAGE
 
-    /* Check arguments */
     HDassert(image);
     HDassert(len > 0);
     HDassert(udata);
     HDassert(udata->common.f);
     HDassert(udata->common.cont_msg_info);
     HDassert(dirty);
 
-    /* Check for partially deserialized object header */
-    /* (Object header prefix will be deserialized if the object header came
-     *  through the 'get_final_load_size' callback and not deserialized if
-     *  the object header is coming from a cache image - QAK, 2016/12/14)
+    /* Check for partially deserialized object header
+     *
+     * The Object header prefix will be deserialized if the object header came
+     * through the 'get_final_load_size' callback and not deserialized if
+     * the object header is coming from a cache image.
      */
     if (NULL == udata->oh) {
         /* Deserialize the object header prefix */
-        if (H5O__prefix_deserialize((const uint8_t *)image, udata) < 0)
+        if (H5O__prefix_deserialize((const uint8_t *)image, len, udata) < 0)
             HGOTO_ERROR(H5E_OHDR, H5E_CANTDECODE, NULL, "can't deserialize object header prefix")
 
-        /* Sanity check */
-        HDassert(udata->oh);
-    } /* end if */
+        if (NULL == udata->oh)
```

Comment:
Or should this go back to being an assert? If we didn't set udata->oh after decoding that seems like something more appropriately caught by an assert rather than a normal error check.

Reply:
Switched back to an assert